dashboards_test.go

```diff
@@ -75,3 +78,109 @@ func (suite *YAxisTestSuite) TestYAxisJSONUnmarshalAutoMinMax() {
 func TestYAxisTestSuite(t *testing.T) {
 	suite.Run(t, new(YAxisTestSuite))
 }
+
+func TestDashboardGetters(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		response, err := ioutil.ReadFile("./tests/fixtures/dashboards_response.json")
+		if err != nil {
+			t.Fatal(err)
+		}
+		w.Write(response)
+	}))
+	defer ts.Close()
+
+	datadogClient := Client{
+		baseUrl:    ts.URL,
+		HttpClient: http.DefaultClient,
+	}
+
+	dashboards, err := datadogClient.GetDashboards()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(dashboards) != 2 {
+		t.Fatalf("expect response with 2 dashboards. Got %d", len(dashboards))
+	}
+
+	d1 := dashboards[0]
+
+	expectedID := 123
+	if id := d1.GetId(); id != expectedID {
+		t.Fatalf("expect ID %d. Got %d", expectedID, id)
+	}
+
+	expectedTitle := "Dashboard 1"
+	if title := d1.GetTitle(); title != expectedTitle {
+		t.Fatalf("expect title %s. Got %s", expectedTitle, title)
+	}
+
+	expectedDescription := "created by user1"
+	if description := d1.GetDescription(); description != expectedDescription {
+		t.Fatalf("expect description %s. Got %s", expectedDescription, description)
+	}
+
+	expectedCreateTime := "2018-10-05T12:32:01.000000+00:00"
+	createTime, ok := d1.GetCreatedOk()
+	if !ok {
+		t.Fatal("expect a time object")
+	}
+
+	if createTime != expectedCreateTime {
+		t.Fatalf("expect create time %s. Got %s", expectedCreateTime, createTime)
+	}
+
+	expectedReadOnly := false
+	if readOnly, ok := d1.GetReadOnlyOk(); !ok || readOnly != expectedReadOnly {
+		t.Fatalf("expect read_only exist and be set to %v. Got exists %v, value %v", expectedReadOnly, ok, readOnly)
+	}
+
+	expectedModified := "2018-09-11T06:38:09.000000+00:00"
+	if modified, ok := d1.GetModifiedOk(); !ok || modified != expectedModified {
+		t.Fatalf("expect modified field to exist and be set to %s. Got exists %v, value %s", expectedModified, ok, modified)
```

Comment:
I see this with the other checks too; if (in this case) `Modified` was *not* set, `GetModifiedOk` will return an empty string and boolean false. In that case this check will fail with (notice value is empty): 

> expect modified field to exist and be set to 2018-09-11T06:38:09.000000+00:00. Got exists false, value

Reply:
fixed!